addons/editor-messages/userscript.js

```diff
@@ -9,9 +9,11 @@ export default async function ({ addon, console, msg }) {
   messageCount.classList.add("sa-editormessages-count");
   messages.appendChild(messageCount);
   const setMessages = async () => {
-    const msgCount = Number(await addon.account.getMsgCount());
-    messageCount.innerText = msgCount;
-    if (msgCount === 0) {
+    const username = await addon.auth.fetchUsername();
+    if (!username) return;
+    const { count } = await (await fetch(`https://api.scratch.mit.edu/users/${username}/messages/count`)).json();
```

Comment:
There's also a double `await`. It might be unnecessary?

Reply:
[‎background/load-addon-manifests.js](https://github.com/ScratchAddons/ScratchAddons/blob/d2a9ce32afd85340702c271d579b7dc6ac455212/background/load-addon-manifests.js#L43) does the same thing, though.

Reply:
Yep - `fetch` returns a promise, and so does `json` (which is only available in the result of the promise, not the promise itself)

Reply:
And: https://github.com/ScratchAddons/ScratchAddons/blob/d2a9ce32afd85340702c271d579b7dc6ac455212/addons/turbowarp-player/userscript.js#L34-L40

Never mind.